### PR TITLE
Fix active account and searchBar observables/subscriptions

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

We should only be subscribing to observables and if we create subscriptions, we should keep a reference and clean them up when not needed anymore (`unsubscribe`). We currently expose the `BehaviourSubject` and subscribe to this and also call unsubscribe. Calling `unsubscribe` on a subject, cancels all subscriptions and not just the one in use.

This is part of https://bitwarden.atlassian.net/browse/TDL-121

## Code changes

- **libs/common/src/abstractions/state.service.ts**: Change type of `activeAccount` to be an `Observable` instead of a `BehaviourSubject`
- **libs/common/src/services/state.service.ts**: 
  - Change type of `activeAccount` to be an `Observable` instead of a `BehaviourSubject`
  - Change `activeAccountSubject` to be used change state (`BehaviourSubject`) internally and not be exposed

- **Create and track the subscription of the activeAccount observable and unsubscribe on `ngOnDestroy`**
  - **libs/angular/src/components/lock.component.ts**: 
  - **apps/desktop/src/app/accounts/lock.component.ts**: 
  - **apps/desktop/src/app/layout/search/search.component.ts**: 

- **apps/desktop/src/app/app.component.ts**: Ensure `OnDestroy` is added which calls `ngOnDestroy`

- **apps/browser/src/popup/app.component.ts**: Fix check for no active accounts to redirect to the login page instead of lock

- **apps/desktop/src/app/layout/search/search-bar.service.ts**: Change subscription handling on SearchBarService 

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/desktop/src/app/layout/search/search-bar.service.ts [hunk 1]
```diff
@@ -8,15 +8,17 @@ export type SearchBarState = {
...
-  searchText = new BehaviorSubject<string>(null);
+  private searchTextSubject = new BehaviorSubject<string>(null);
+  searchText = this.searchTextSubject.asObservable();
 
   private _state = {
     enabled: false,
     placeholderText: "",
   };
 
   // tslint:disable-next-line:member-ordering
```
Comment (MGibson1) [inline apps/desktop/src/app/layout/search/search-bar.service.ts:19]: Can delete this linter hint

File: apps/desktop/src/app/layout/search/search-bar.service.ts [hunk 2]
```diff
@@ -8,15 +8,17 @@ export type SearchBarState = {
 
 @Injectable()
 export class SearchBarService {
-  searchText = new BehaviorSubject<string>(null);
+  private searchTextSubject = new BehaviorSubject<string>(null);
+  searchText = this.searchTextSubject.asObservable();
```
Comment (MGibson1) [inline apps/desktop/src/app/layout/search/search-bar.service.ts:12]: Please follow convention to name observables with a `$` suffix (https://angular.io/guide/rx-library#naming-conventions-for-observables)

File: apps/desktop/src/app/layout/search/search-bar.service.ts [hunk 3]
```diff
@@ -8,15 +8,17 @@ export type SearchBarState = {
...
 
   private _state = {
     enabled: false,
     placeholderText: "",
   };
 
   // tslint:disable-next-line:member-ordering
-  state = new BehaviorSubject<SearchBarState>(this._state);
+  private stateSubject = new BehaviorSubject<SearchBarState>(this._state);
+  state = this.stateSubject.asObservable();
```
Comment (MGibson1) [inline apps/desktop/src/app/layout/search/search-bar.service.ts:21]: Please name with `$` suffix

File: libs/common/src/abstractions/state.service.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ import { SendView } from "../models/view/sendView";
 
 export abstract class StateService<T extends Account = Account> {
   accounts: BehaviorSubject<{ [userId: string]: T }>;
-  activeAccount: BehaviorSubject<string>;
+  activeAccount: Observable<string>;
```
Comment (MGibson1) [inline libs/common/src/abstractions/state.service.ts:29]: Please name with `$` suffix

File: libs/common/src/services/state.service.ts [hunk 5]
```diff
@@ -55,8 +55,9 @@ export class StateService<
 > implements StateServiceAbstraction<TAccount>
 {
   accounts = new BehaviorSubject<{ [userId: string]: TAccount }>({});
-  activeAccount = new BehaviorSubject<string>(null);
+  private activeAccountSubject = new BehaviorSubject<string>(null);
   activeAccountUnlocked = new BehaviorSubject<boolean>(false);
+  activeAccount = this.activeAccountSubject.asObservable();
```
Comment (MGibson1) [inline libs/common/src/services/state.service.ts:60]: Please name with `$` suffix

**Review by MGibson1 — CHANGES_REQUESTED** [review]
This looks good, just a few convention book keeping items 📚

File: apps/desktop/src/app/layout/search/search-bar.service.ts (hunk 2 above)
Comment (djsmith85) [reply apps/desktop/src/app/layout/search/search-bar.service.ts]: Thanks, looks better following the convention.

**Comment by MGibson1** [comment]
relates to https://github.com/bitwarden/clients/issues/2905

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**